### PR TITLE
5032 -  Add ids breadcrumb Component

### LINKED ISSUE #137: IdsBreadCrumb: Add new web component
**Describe the bug**

Bring over the current "breadcrumb" component.  See if we can simplify and improve the code at all while improving as we go. Current breadcrumb component https://master-enterprise.demo.design.infor.com/components/breadcrumb

```
<ids-bread-crumb truncate="true">
 <ids-hyperlink href="#">Home</ids-hyperlink>
 <ids-hyperlink href="#">Second Item</ids-hyperlink>
 <ids-hyperlink href="#">Third Item </ids-hyperlink>
 <ids-hyperlink href="#"  current="true">Fourth Item</ids-hyperlink>
</ids-bread-crumb>
```

**Phase One*
- Supports hyperlinks
- Supports static text/clickable (ids-text vs ids-hyperlink)
- Supports disabled (use disabled on ids-hyperlink and ids-text)
- Supports API to add and remove breadcrumbs
- Currently selected state ( current="true" - bold)
- Supports being in a toolbar where the title is
- For accessible markup see https://www.w3.org/TR/wai-aria-practices-1.1/examples/breadcrumb/index.html

**Phase Two (Future)**
- Truncation on mobile/less wide (Not needed for MVP) 
- Placement in a header (when ids-header is done)

**Additional context**
- 100% test coverage
- DOCS, CHANGE log ect. 
- https://github.com/infor-design/enterprise-wc/blob/master/doc/CHECKLIST.MD
- themes 

See components like card, tag, badge, hyperlink as reference

### PR DESCRIPTION
**Pull request by chrisakinrinade**
**Explain the details for making this change. What existing problem does the pull request solve?**
Adding IdsBreadcrumb component to the web component repo. It uses IdsHyperlinks as inner components.

**Related github/jira issue (required)**:
Fixes #137 

**Steps necessary to review your pull request (required)**:
Pull this branch, build, and run the app
Go to http://localhost:4300/ids-breadcrumb
Check the code implementations and functional tests.

**Included in this Pull Request**:
- [x] An e2e or functional test for the bug or feature.
- [x] A note to the change log.

### CODE REVIEW

File: app/ids-breadcrumb/add-remove.html [hunk 1]
```diff
@@ -0,0 +1,22 @@
+{{> ../layouts/head.html }}
+
+<ids-layout-grid>
+  <ids-text font-size="12" type="h1">BREADCRUMBS</ids-text>
```
Comment (tmcconechy) [inline app/ids-breadcrumb/add-remove.html:4]:
```suggestion
  <ids-text font-size="12" type="h1">Add/Remove Breadcrumbs</ids-text>
```

File: app/ids-breadcrumb/add-remove.html [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+    <ids-hyperlink color="unset" font-size="20" href="#">Large Breadcrumb</ids-hyperlink>
+    <ids-hyperlink color="unset" disabled href="#">Disabled Breadcrumb</ids-hyperlink>
+  </ids-breadcrumb>
+</ids-layout-grid>
+<br /><br />
+<ids-layout-grid>
+  <ids-checkbox label="Disabled"></ids-checkbox>
+</ids-layout-grid>
+<ids-layout-grid cols="10" gap="md">
+  <ids-button type="primary">Add Link</ids-button>
```
Comment (tmcconechy) [inline app/ids-breadcrumb/add-remove.html:18]:
```suggestion
  <ids-button type="secondary">Add Link</ids-button>
```

File: src/ids-breadcrumb/README.md [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+## Description
+
+Displays the current view and all parent views. A user can navigate between views. Best for presenting hierarchical paths in a system.
+
+## Use Cases
+
+- Use breadcrumb in a tree-like page navigation to show the path to the current page.
+
+## Terminology
+/*
```
Comment (tmcconechy) [inline src/ids-breadcrumb/README.md:12]: Dont think these comments are needed?

File: src/ids-breadcrumb/ids-breadcrumb.d.ts [hunk 4]
```diff
@@ -0,0 +1,6 @@
+// Ids is a JavaScript project, but we define TypeScript declarations so we can
+// confirm our code is type safe, and to support TypeScript users.
+
+/** A segment of text with standardized settings, theming and fonts */
+export default class IdsBreadcrumb extends HTMLElement {
```
Comment (tmcconechy) [inline src/ids-breadcrumb/ids-breadcrumb.d.ts:5]: Can you add all the types for settings? At least mode and theme i think if no other settings?

File: src/ids-breadcrumb/ids-breadcrumb.js [hunk 5]
```diff
@@ -0,0 +1,87 @@
...
+  mix,
+} from '../ids-base';
+
+// Import Mixins
+import { IdsEventsMixin, IdsThemeMixin } from '../ids-mixins';
+
+import styles from './ids-breadcrumb.scss';
+
+/**
+ * @type {IdsBreadcrumb}
```
Comment (tmcconechy) [inline src/ids-breadcrumb/ids-breadcrumb.js:14]:
```suggestion
 *  IDS Breadcrumb Component
 * @type {IdsBreadcrumb}
```

File: src/ids-breadcrumb/ids-breadcrumb.js [hunk 6]
```diff
@@ -0,0 +1,87 @@
...
+      <div class="ids-breadcrumb">
+        <nav>
+          <slot></slot>
+        </nav>
+      </div>`;
+  }
+
+  /**
+   * Appends an individual breadcrumb to the end of the stack
+   * @param {Element} breadcrumb The HTML element with which to add
```
Comment (tmcconechy) [inline src/ids-breadcrumb/ids-breadcrumb.js:56]:
```suggestion
   * @param {Element} breadcrumb The HTML element with which to add
   * @private
```

And could make it #push (private class)

File: src/ids-breadcrumb/ids-breadcrumb.js [hunk 7]
```diff
@@ -0,0 +1,87 @@
...
+      breadcrumb.setAttribute('font-size', 14);
+    }
+    this.appendChild(breadcrumb);
+  }
+
+  /**
+   * Removes the last breadcrumb from the stack.
+   * @returns {Element | null} The removed element
+   */
+  pop() {
```
Comment (tmcconechy) [inline src/ids-breadcrumb/ids-breadcrumb.js:75]: also private?

**Review by tmcconechy — CHANGES_REQUESTED** [review]
Looking good ! A few small edits and a couple things to improve it

1. Its missing focus states -> try tabbing into the bread crumb -> you might just need href="#" in the markup
2. Along those lines trying http://localhost:4300/ids-breadcrumb/add-remove and i see a mix of links with underline or not? Is that related to the href? And maybe remove the large example
3. http://localhost:4300/ids-breadcrumb/standalone-css doesnt seem to look the same for some reason?

**Comment by chrisakinrinade** [comment]
> Looking good ! A few small edits and a couple things to improve it
> 
> 1. Its missing focus states -> try tabbing into the bread crumb -> you might just need href="#" in the markup
> 2. Along those lines trying http://localhost:4300/ids-breadcrumb/add-remove and i see a mix of links with underline or not? Is that related to the href? And maybe remove the large example
> 3. http://localhost:4300/ids-breadcrumb/standalone-css doesnt seem to look the same for some reason?
> 4. And maybe comment out the coverage issue

@tmcconechy 
1. Should crumbs without hrefs be focusable?
2. The underline is an issue w/ the hyperlink -- the css for it just hasn't been done yet. Do you think that should be fixed with this ticket? And np, I'll get rid of the large example.
3. The sizing was done w/ js. I'll see if there's a workaround I can do about that.

**Comment by tmcconechy** [comment]
1. I think it would be ok if they are not focusable if no href (as its just static text) but the color should look the same as before
2. If its easy? As it looks strange if they are mixed but if its only when adding/removing or its complicated then we can leave it for now
3. 👍🏻

File: src/ids-breadcrumb/ids-breadcrumb.js (hunk 6 above)
Comment (chrisakinrinade) [reply src/ids-breadcrumb/ids-breadcrumb.js:47]: Actually I intended push and pop to be public. That's how people would interface with the breadcrumbs.

**Review by tmcconechy — DISMISSED** [review]

**Review by deep7102 — COMMENTED** [review]
Looks good, few things see
http://localhost:4300/ids-breadcrumb
1) hover styles (is should as 4.x)
2) is last one `Current Item` should be hyperlink or just text?

http://localhost:4300/ids-breadcrumb/standalone-css
1) Disabled (keyboard has focus state)

**Comment by tmcconechy** [comment]
@deep7102 I fixed those points just now. But note that the style in the old system is incorrect on hover as it changes blue color. So i left that off it just shows the underline.

**Review by deep7102 — APPROVED** [review]
🍌 Looks good, Thanks!

**Pull request merged**